src/app/profile/profile.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, FormArray, Validators } from '@angular/forms';
+import { customEmailValidator } from './validators/email.validator';
+
+import { Profile } from './profile.model';
+import { ProfileService } from '../profile.service';
+
+@Component({
+  selector: 'app-profile',
+  templateUrl: './profile.component.html',
+  styleUrls: ['./profile.component.css']
+})
+
+export class ProfileComponent implements OnInit {
+
+  profiles: any;
+
+  profileForm!: FormGroup;
+
+  firstName = new FormControl('', Validators.required);
+  lastName = new FormControl('', Validators.required);
+  email = new FormControl('', [ Validators.required, customEmailValidator() ]);
+
+  constructor(private profileService: ProfileService) { }
+
+  getProfileData(): void {
```

Comment:
I would pull getProfileData() below ngOnInit.  I prefer to keep the ngOnInit and lifecycle events under the constructor for organization.    Also, getProfileData() or any other function not meant for the template should be marked as private.

Reply:
got it, I was unsure of my arrangement of things, so thanks for letting me know what you do.